source/slang/modifier-defs.h

```diff
@@ -328,6 +327,9 @@ SIMPLE_SYNTAX_CLASS(FlattenAttribute, Attribute)            // `[flatten]`
 SIMPLE_SYNTAX_CLASS(ForceCaseAttribute, Attribute)          // `[forcecase]`
 SIMPLE_SYNTAX_CLASS(CallAttribute, Attribute)               // `[call]`
 
+// [[vk_push_constant]] [[push_constant]]
+SIMPLE_SYNTAX_CLASS(PushConstantAttribute, Attribute)
```

Comment:
One concern here is that we have no way of differentiating the `[[push_constant]]` case from the `[[vk::push_constant]]` case. If we are going to have both options then I'd advocate for a hierarchy of three classes:

* `PushConstantAttributeBase` - base class for all push-constant-buffer attributes
  * `VKPushConstantAttribute` - specific case for `[[vk::push_constant]]`
  * `PushConstantAttribute` - specific case for `[push_constant]`

This lets code test for the general case, the VK-specific case, or the catch-all case pretty easily. Although I guess this gets back to the problem of needing a general way to differentiate target-specific vs. target-independent attributes.

I'm not too concerned about this for now, since the 99% use case will just be `[[vk::push_constant]]`, and I'd like to consider alternative approaches for a catch-all version (e.g., using `PushConstant<T>` in place of `ConstantBuffer<T>` as a declaration of intent).